src/librustc/ty/context.rs

```diff
@@ -482,6 +482,23 @@ impl<'tcx> TypeckTables<'tcx> {
         }
     }
 
+    pub fn opt_type_dependent_def(&self, id: HirId) -> Option<&Def> {
```

Comment:
you could return `Def` instead of `&Def` (by using `.copied()` inside this function)

Reply:
This should allow you to save all the `.cloned()` calls after the `opt_type_dependent_def` calls.